src/e-build.js

```diff
@@ -85,6 +86,9 @@ program
         ? target === targets.chromium
         : targets.default === targets.chromium;
       if (process.platform === 'darwin' && !isChromium) {
+        // Setting DEVELOPER_DIR lets us override the Xcode path without requiring sudo
+        process.env.DEVELOPER_DIR = XcodePath;
```

Comment:
Can we set this inside of our `spawn` and `exec` env helper so that it applies for _all_ commands not just `e build`. E.g. `e test` technically uses xcode, as do various `e d *` commands

Reply:
Good point about setting for other commands besides `e build`. Can you clarify what you mean by `spawn` and `exec` env helper? I found the `runSpecRunner` helper for `e-test`, and the `execFileSync` and `spawnSync` in `depot-tools.js`. Are those the ones you're talking about, or is there some central place I should add this?

I could also probably just set `process.env.DEVELOPER_DIR` in `src/e` but then that would apply super broadly to every subcommand. Is that a good idea, or would there be bad side effects from doing that so broadly?

Reply:
There's a centralized spot for `execFileSync` and `spawnSync` helpers where we generate env vars for sub procs

https://github.com/electron/build-tools/blob/main/src/utils/depot-tools.js#L74-L84